### DISCUSSION TITLE
destination modified

### DISCUSSION THREAD

**Discussion by danboid** [psy0rz/zfs_autobackup#230] [Q&A] [answered]
It seems zfs_autobackup is too safe for my liking.

I'm getting:

cannot receive incremental stream: destination has been modified since most recent snapshot

On my main dataset. Yes, the target has changed because I logged into it but I still want it synced.

I looked at the solutions to this on the the wiki and I didn't like any of them. I don't want to make the target dataset read only or roll it back. Is using --force my only choice, will that even fix it?

Basically I'd like zfs-autobackup to not care about the state of the target dataset. If the target dataset doesn't exist any more, it should re-create it if the source exists. If the target has changed, those changes should be overwritten by the sync from the source. Will --force make it work like this? I want all datasets synced like this every update, no questions asked.

**Comment by psy0rz** [comment, marked as answer]
You did something that made a change on the target dataset. (even just mounting it can do this)

Either --force and --rollback would "fix" this issue, by throwing away those changes.

More is explained here: https://github.com/psy0rz/zfs_autobackup/wiki/Mounting

**Reply by danboid** [reply]
Yes, adding --force has fixed the sync but zfs-autobackup still prints an error for the affected dataset during the Sync step when run in verbose mode but I can live with that.

**Reply by psy0rz** [reply]
what error does it print?

**Reply by danboid** [reply]
I'm not running --force again until you answer my questions in the "Uh oh!" discussion but the error was the same as before, the "cannot receive incremental stream: destination has been modified since most recent snapshot" one.
